skaffold/src/test/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationsTest.kt

```diff
@@ -0,0 +1,116 @@
+import com.intellij.openapi.project.guessProjectDir
+import com.intellij.util.ThrowableRunnable
+import io.mockk.CapturingSlot
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.mockkStatic
+import io.mockk.slot
+import org.jdom.Element
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
+
+/** Unit tests for [SkaffoldCommandLineState] */
+class SkaffoldRunConfigurationsTest {
+    @get:Rule
+    val containerToolsRule = ContainerToolsRule(this)
+
+    private lateinit var skaffoldCommandLineState: SkaffoldCommandLineState
+    private lateinit var abstractSkaffoldRunConfiguration: AbstractSkaffoldRunConfiguration
+
+    @MockK
+    private lateinit var mockExecutionEnvironment: ExecutionEnvironment
+    @MockK
+    private lateinit var mockRunnerSettings: RunnerAndConfigurationSettings
+    @MockK
+    private lateinit var mockDevConfiguration: SkaffoldDevConfiguration
+    @MockK
+    @TestService
+    private lateinit var mockSkaffoldExecutorService: SkaffoldExecutorService
+    @MockK
+    @TestService
+    private lateinit var mockPluginInfoService: PluginInfo
+
+    private val skaffoldSettingsCapturingSlot: CapturingSlot<SkaffoldExecutorSettings> = slot()
+
+    @Before
+    fun setUp() {
+        every {
+            mockExecutionEnvironment.runnerAndConfigurationSettings
+        } answers { mockRunnerSettings }
+
+        every { mockRunnerSettings.configuration } answers { mockDevConfiguration }
+        // pass project into the CLI state
+        every { mockExecutionEnvironment.project } answers {
+            containerToolsRule.ideaProjectTestFixture.project
+        }
+
+        // CommandLineState calls this static method, needs to be mocked
+        mockkStatic(SearchScopeProvider::class)
+        every { SearchScopeProvider.createSearchScope(any(), any()) } answers { mockk() }
+
+        // Skaffold executor answer mocks
+        val mockSkaffoldProcess: SkaffoldProcess = mockk(relaxed = true)
+        every {
+            mockSkaffoldExecutorService.executeSkaffold(capture(skaffoldSettingsCapturingSlot))
+        } answers { mockSkaffoldProcess }
+    }
+
+
+    @Test
+    fun `A runtime warning is shown if skaffold isn't in the system PATH`() {
+
+        val testObj = mockk<AbstractSkaffoldRunConfiguration.Companion>()
+
+        every { testObj.skaffoldPath } returns null
+
+        every { testObj.checkConfiguration(mockExecutionEnvironment.project) } throws ExecutionException("skaffold.not.in.system.error")
+    }
+
+    @Test
+    fun `A runtime warning is not shown if skaffold is in the system PATH`() {
+
+        val testObj = mockk<AbstractSkaffoldRunConfiguration.Companion>()
+
+        every { testObj.skaffoldPath } returns Paths.get("skaffold")
+
+        testObj.checkConfiguration(mockExecutionEnvironment.project)
+
+    }
+}
```

Comment:
I think we add a newline at the end of our files - I believe this is a common GitHub practice.